buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java

```diff
@@ -34,59 +37,59 @@
+ *   <li>the unreleased <b>maintenance</b>, M-1.d.e ( d &gt; 0, e &gt; 0) on the `(M-1).d` branch
  * </ul>
- * In addition to these, there will be a fifth one when a minor reaches feature freeze, we call this the <i>staged</i>
- * version:
+ *
+ * In addition to these, there will be a fifth one when a minor reaches feature freeze, we call this
+ * the <i>staged</i> version:
+ *
  * <ul>
- * <li>the unreleased <b>staged</b>, M.N-2.0 (N &gt; 2) on the `M.(N-2)` branch</li>
+ *   <li>the unreleased <b>staged</b>, M.N-2.0 (N &gt; 2) on the `M.(N-2)` branch
  * </ul>
  *
- * Each build is only concerned with versions before it, as those are the ones that need to be tested
- * for backwards compatibility. We never look forward, and don't add forward facing version number to branches of previous
- * version.
+ * Each build is only concerned with versions before it, as those are the ones that need to be
+ * tested for backwards compatibility. We never look forward, and don't add forward facing version
+ * number to branches of previous version.
  *
- * Each branch has a current version, and expected compatible versions are parsed from the server code's Version` class.
- * We can reliably figure out which the unreleased versions are due to the convention of always adding the next unreleased
- * version number to server in all branches when a version is released.
- * E.x when M.N.c is released M.N.c+1 is added to the Version class mentioned above in all the following branches:
- *  `M.N`, `M.x` and `master` so we can reliably assume that the leafs of the version tree are unreleased.
- * This convention is enforced by checking the versions we consider to be unreleased against an
- * authoritative source (maven central).
- * We are then able to map the unreleased version to branches in git and Gradle projects that are capable of checking
+ * <p>Each branch has a current version, and expected compatible versions are parsed from the server
+ * code's Version` class. We can reliably figure out which the unreleased versions are due to the
+ * convention of always adding the next unreleased version number to server in all branches when a
+ * version is released. E.x when M.N.c is released M.N.c+1 is added to the Version class mentioned
+ * above in all the following branches: `M.N`, `M.x` and `master` so we can reliably assume that the
+ * leafs of the version tree are unreleased. This convention is enforced by checking the versions we
+ * consider to be unreleased against an authoritative source (maven central). We are then able to
+ * map the unreleased version to branches in git and Gradle projects that are capable of checking
  * out and building them, so we can include these in the testing plan as well.
  */
 public class BwcVersions {
 
-    private static final Pattern LINE_PATTERN = Pattern.compile(
-        "\\W+public static final Version V_(\\d+)_(\\d+)_(\\d+)(_alpha\\d+|_beta\\d+|_rc\\d+)? .*"
-    );
+    private static final Pattern LINE_PATTERN =
```

Comment:
I think this is an example where the formatter had no business and made things worse.

Reply:
I don't exactly dislike what GJF did, but it's interesting that e.g. Prettier formats code more like we do at the moment. I wonder how far [prettier-java](https://github.com/jhipster/prettier-java) will get - not that I want to add a JS build dependency to a Java project 😅